### DISCUSSION TITLE
add option to delete multiple scores at a time

### DISCUSSION THREAD

**Discussion by nanashi-1** [ppy/osu#19498] [Ideas]
In the current osu!lazer version (2022.731.1), osu!lazer does not allow deleting multiple scores at a time.

Below is a way to delete scores.

[Screencast from 07-31-2022 06:01:21 PM.webm](https://user-images.githubusercontent.com/54083498/182022973-a0752214-5892-4a7d-9ce2-5ca99990f350.webm)

I suggest adding a feature similar to GMail's Selection tool.

![Screenshot from 2022-07-31 19-06-08](https://user-images.githubusercontent.com/54083498/182023256-6c45852b-8dbd-4843-8581-cd43923bbbdc.png)

**Comment by peppy** [comment]
To help figure out a good UX for this, what is the general case you'd be wanting to delete multiple scores? All scores with a certain grade? All scores below a certain score?

**Reply by nanashi-1** [reply]
Would be good to remove very old scores like from a month ago and very low scores.

**Reply by BsRebornPhoenix** [reply]
> To help figure out a good UX for this, what is the general case you'd be wanting to delete multiple scores? All scores with a certain grade? All scores below a certain score?

is this still coming to lazer? its hella annoying deleting replays 1 by 1

**Reply by Tenexxt** [reply]
for a start, when pressing shift and clicking on a score, it could become selected. this way you would select scores and then after right clicking on any selected score you would get a option to delete the selected scores

along with that, there could be a local leaderboard filter button, that would open a small menu where you select the criteria of shown scores (such as grade, accuracy, mods, time set etc.) and a "select all scores button", which would essencially make deleting specific scores very quick

**Reply by rvnexe** [reply]
The best UX for this may be the behaviour like in windows explorer;
Where you can RightClick + Drag to **select multiple files** and **open context menu** simultanously.
While rightclick dragging doesn't have any purpose in osu! _(unlike some games, for example valorant where rightclick dragging works as a scrolling method)_ it would be so useful to implement.

**Reply by rvnexe** [reply]
^ this way, you can also Export multiple scores replays to files

**Reply by Tenexxt** [reply]
actually, when you hold right mouse button around the beatmap carousel, it will snap the scroll to the cursor position, so you can use this to travel across song select very fast

**Reply by rvnexe** [reply]
It has no purpose in ur score list I mean*
